io.openems.shared.influxdb/src/io/openems/shared/influxdb/proxy/QueryProxy.java

```diff
@@ -132,6 +132,27 @@ public abstract SortedMap<ZonedDateTime, SortedMap<ChannelAddress, JsonElement>>
 			Resolution resolution //
 	) throws OpenemsNamedException;
 
+	/**
+	 * Queries the last value of a channel, e.g. an engerycounter.
+	 * 
+	 * @param influxConnection a Influx-Connection
+	 * @param bucket           the bucket name; 'database/retentionPolicy' for
+	 *                         InfluxDB v1
+	 * @param measurement      the influx measurement
+	 * @param influxEdgeId     the Edge-ID
+	 * @param channel          the {@link ChannelAddress}
+	 * @return the query result
+	 * @throws OpenemsNamedException on error
+	 */
+	public abstract SortedMap<ZonedDateTime, SortedMap<ChannelAddress, JsonElement>> queryLastData(
```

Comment:
i think this method is pretty similar to the one below `queryFirstValueBefore` (the name of the method may be a little bit off) can we somehow merge those?

Reply:
yeah...no...maybe... queryFirstValueBefore is used by many modules. As I am not a real java programmer I don´t know if I can provide this method. 
What about a compromise: you check in my implementation and I´ll put "queryFirstValueBefore" on my ToDo list?

Reply:
only because its used in other models doesnt mean you cant use it or have to provide it only the flux implementation is missing currently but i could help you with that.

I just dont want to add a method which already exists. It would just lead to more confusion i guess and its also harder to maintain. 

What we also can do is to keep the method in the InfluxConnector and call the `queryFirstValueBefore` method inside but we pass the one channel as a `Set` and just the current time. It would look like this i guess:
```
this.queryProxy.queryFirstValueBefore(this.bucket, this.getInfluxConnection(), measurement, influxEdgeId,
				ZonedDateTime.now(), Set.of(channelAddress));
```

Reply:
Hi Michael,
I´ve re-coded the pull-request. 
- InfluxConnector: queryLastData makes use of the queryFirstValueBefore-method
- FluxProxy: added buildFetchFirstValueBefore and convertFirstValueBeforeQueryResult
- TimedataInfluxDbImpl: added method getLatestValue